firmware/src/networking/protocol/mod.rs

```diff
@@ -0,0 +1,68 @@
+//! THIS MODULE IS WIP AND NOT FINISHED YET
+//!
+//! Manages the slimevr firmware protocol. All (de)serialization and state associated with
+//! the protocol happens in here.
+#![allow(dead_code)]
+
+mod serialize;
+
+use crate::{Quat, Unreliable};
+
+use embassy_sync::{
+	blocking_mutex::raw::NoopRawMutex,
+	channel::{Channel, Receiver, Sender},
+};
+use firmware_protocol::{CbPacket, SbPacket};
+
+pub struct ImuHandle {
+	signal: &'static Unreliable<Quat>,
+}
+impl ImuHandle {
+	/// Send the latest quaternion. May drop the data.
+	pub fn send_quat(&mut self, q: Quat) {
+		self.signal.signal(q)
+	}
+}
+
+pub struct Protocol {
+	quat_signal: &'static Unreliable<Quat>,
+	sb: Channel<NoopRawMutex, SbPacket, 1>, // TODO: Should these actually be channels?
+	cb: Channel<NoopRawMutex, CbPacket, 1>,
+}
+impl Protocol {
+	pub fn new(quat_signal: &'static Unreliable<Quat>) -> (Protocol, ImuHandle) {
+		let self_ = Self {
+			quat_signal,
+			sb: Channel::new(),
+			cb: Channel::new(),
+		};
+		(
+			self_,
+			ImuHandle {
+				signal: quat_signal,
+			},
+		)
+	}
+
+	/// Gets a `Receiver` for serverbound packets that ought to be sent.
+	pub fn serverbound(&self) -> Receiver<'_, NoopRawMutex, SbPacket, 1> {
```

Comment:
`Channel` is not a PubSub (and we do not need it to be) so creating multiple receivers will lead to incorrect behavior.

Reply:
I opted to expose the actual receiver and producer for now, since it had a richer api and I didn't feel like duplicating it. As comented before, I'm not even sure if these should be channels yet, since both ends reside in the same async task. I need to experiment more. 

However, the `Protocol` struct is owned by the networking task, so the networking task can decide what to do with that receiver. I don't see an issue with this function for that reason.